elytron/WFLY-16306-keycloak-saml-feature-pack.adoc

```diff
@@ -0,0 +1,331 @@
+and moved to a separate repo. We don't have details yet on when this will happen. We'll likely need to add our feature pack to the
+existing Keycloak project initially and then it will get moved to the new project with the rest of the adapter code.
+
+A new Galleon feature pack will be added to the Keycloak project with the group ID, `org.keycloak`, and artifact ID, `keycloak-saml-adapter-galleon-pack`.
+This new feature pack will contain three layers called `keycloak-saml`, `keycloak-client-saml`, and `keycloak-client-saml-ejb`.
+
+When this feature pack and its layers are provisioned to a WildFly installation, the end result will mostly match the results
+of manually installing the Keycloak Elytron SAML adapter using the instructions linked above.
+
+In particular, when the `keycloak-saml` layer is provisioned, the following steps will occur:
+
+ # Install a set of modules under `modules/system/add-ons`
+ # Register the `org.keycloak.keycloak-saml-adapter-subsystem` extension.
+ # Add an empty `urn:jboss:domain:keycloak:1.4` subsystem.
+
+The `keycloak-client-saml` layer will depend on the `keycloak-saml` layer and when provisioned, the following additional steps will occur:
+
+ # Add the required resources to the `elytron` subsystem to result in an `http-authentication-factory` and `security-domain` supporting SAML being available.
+ # Update the `application-security-domain` mapping in the `undertow` subsystem to map the default security domain to the new `http-authentication-factory`.
+
+The `keycloak-client-saml-ejb` layer will depend on the `keycloak-client-saml` layer and when provisioned, the following additional steps will occur:
+
+ # Update the `application-security-domain` mapping in the `ejb3` subsystem to map the default security domain to the new `KeycloakDomain`.
+
+Note that the existing CLI scripts for the SAML adapter and the existing feature pack for OIDC only update the `application-security-domain` mapping in the
+`undertow` subsystem. However, the CLI scripts for the OIDC adapter also update the `application-security-domain`
+mapping in the EJB3 subsystem. For this reason, we have added the `keycloak-client-saml-ejb` layer here for users
+that need to propagate their identities to EJBs as well.
+
+The CLI scripts that would have been added via the manual installation steps won't be added to the resulting installation
+since the required resources will be added already during provisioning.
+
+Note that Keycloak and WildFly have different release cadences. This means anyone using Galleon to keep their WildFly
+installation up to date should be able to receive updates to the core WildFly installation independently of the new
+Keycloak feature pack. Having the new Keycloak feature pack installed should not prevent the server from being updated
+once later WildFly versions are available. Similarly, WildFly releases should not trigger the need for an update to Keycloak.
+
+==== Bare Metal vs. Cloud Use Cases
```

Comment:
What are possibilities to configure the `keycloak-client-saml` for bare metal? 

In case of the OpenShift I am expecting usage of deployment variables SSO_XXX to configure it. 

For bare matel is CLI script only one way?

Reply:
For bare metal, either a custom CLI script could be used to configure the `keycloak-saml` subsystm or deployment configuration could be used.

Reply:
S2I process also support automatic client registration in realm of RH-SSO. Is supported this feature also for bare metal? or it is only S2I specific feature?

Reply:
That's an s2i specific feature.

Reply:
this might be relevant here: in general, we use layers `cloud-default-config`+`keycloak-saml` when using `wildfly-maven-plugin` for s2i (this automatically registers the SAML deployment into EAP configuration) .... we use the other layers for everything else (e.g. `web-server`+`keycloak-client-saml` - this does NOT register the SAML deployment into EAP configuration and that's why some user cli script like `/subsystem=keycloak-saml/secure-deployment=YOUR-WAR.war/:add ...` is needed - a `keycloak-saml.xml` in the deployment works too)

Reply:
I don't see the role of Galleon layers there. You could use cloud-server and keycloak-saml, cloud-default-config is not mandatory at all. Could you develop what cloud-default-config brings?